### PR TITLE
Documentation for Anomaly types + use "ugly" sampling frequency

### PR DESCRIPTION
**Pull request by breznak**
- new doc for subfolders of anomaly dataset with explanations
- use (and explanation why it's important!) fractional (1.001) sampling frequency (FS)
  - datasets and figures rerun with the new FS

### CODE REVIEW

**Comment by breznak** [comment]
thanks @wattik ! is this all for the "documentation" batch? 
I'll have just wording comments to the readmes

File: .gitignore [hunk 1]
```diff
@@ -34,3 +34,15 @@ nosetests.xml
 .mr.developer.cfg
 .project
 .pydevproject
+
+opf/.idea/.name
```
Comment (breznak) [inline .gitignore]: is this data for some IDE you are using? Can we mask all the `.idea` files?

File: data/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Datasets (This section is work in progress.)
+
+## Anomalies datasets
```
Comment (breznak) [inline data/README.md]: Anomaly datasets

File: data/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+There are several types of anomalies in data streams. We categorize them in the following way:
+- neatData – clean data sets generated with no anomalies included. Eg. sine, constant, spike train.
```
Comment (breznak) [inline data/README.md]: - can use some "GH styling" like, `neatData/`
- remove the word "included" there

File: data/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+- corruptedData – these data sets has been taken from the previous group but then corrupted by deterministic anomalies. Subfolders comprise datasets grouped by the category. Such as: point anomaly, section anomaly, etc.
```
Comment (breznak) [inline data/README.md]: Maybe "Subfolders include specific anomaly types (categories) which we have identified, such as: point anomaly, section anomaly, etc. " ?

File: data/README.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+- phase – a real number addressing in which phase the given periodic function is. For instance, 'phase' column would include numbers from 0 to 2*pi, following the phase of sinus. 
+- has_anomaly – boolean, if i-th sample contains anomaly, 'anomaly at i-th row' = 1, otherwise 0.
+
+Results are cvs files as well. They contain additional columns: anomaly_score, anomaly_likelihood.
+- anomaly_score – raw anomaly score generated by HTM core
+- anomaly_likelihood – likelihood generated by HTM core
+
+## Why we use a strange f = 1.001 Hz?
+
+We use data that are generated by a single frequency function, eg. 1 Hz sine or a spike train. The frequency of the function is f. Then, we sample the continuous function with the sampling frequency fs. Having considered the sampling theorem, we get a ratio K = fs/f > 2. If the ratio K is a natural number, what we get is a sequence of K numbers which are repeated exactly the same over and over again.
```
Comment (breznak) [inline data/README.md]: just a nit, consider using `f`, `fs` (the quotes `) to highlight code name or specific variables

File: data/README.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+Results are cvs files as well. They contain additional columns: anomaly_score, anomaly_likelihood.
+- anomaly_score – raw anomaly score generated by HTM core
+- anomaly_likelihood – likelihood generated by HTM core
+
+## Why we use a strange f = 1.001 Hz?
+
+We use data that are generated by a single frequency function, eg. 1 Hz sine or a spike train. The frequency of the function is f. Then, we sample the continuous function with the sampling frequency fs. Having considered the sampling theorem, we get a ratio K = fs/f > 2. If the ratio K is a natural number, what we get is a sequence of K numbers which are repeated exactly the same over and over again. 
+
+The algorithm then does not create a model of the input data (sine function eg.), but it creates a model of a numbers series. That is a different task and, in fact, immensely easier. If we ensure an ugly real number, then there is a great chance we do not have any numbers series in the data. As a result, the algorithm really develops the model of the function.
```
Comment (breznak) [inline data/README.md]: very nice paragraph! :+1: 
maybe add? "... This behavior (with "ugly" `fs`) correctly mimics the use on real-world streaming data, such as in [River-View](https://github.com/nupic-community/river-view/) which is the focus of most HTM use-cases. 

We have discovered this issue to be a common problem in a lot of NuPIC's examples and even tests, it deserves proper validation and possible further steps (updating NuPIC test-cases, ...)"

Comment (breznak) [inline data/README.md]: @wattik create a separate issue to track the "ugly FS" idea?

File: data/datasets/corruptedData/pointAnomaly/amplitude/README.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+The density of an anomaly is considered as probability p that an anomaly
+would occur in a sample. The general density, say in a period, has
+bionomial distribution with a parameter p. Since it is computationally
+costly to work with binomial distribution, we decided to use a
+probability of the single error instead.
+
+The Good values of the anomaly density are in the range from 0.001 to 0.1. Mind the fact that it heavily depends on the character of data.
+
+If you are not familiar with what we have here, check the _EXAMPLE.eps file.
```
Comment (breznak) [inline data/datasets/corruptedData/pointAnomaly/amplitude/README.md:13]: could we use a PNG or something and include it in the readme directly?

File: data/datasets/corruptedData/sectionAnomaly/README.md [hunk 8]
```diff
@@ -0,0 +1,8 @@
+# Corrupted data - section anomaly
+
```
Comment (breznak) [inline data/datasets/corruptedData/sectionAnomaly/README.md:2]: add better explanation. 
"Unlike point-anomaly, in this category we focus on continuous intervals where there is a lot of anomalies, then is not, then is again etc."

File: data/datasets/corruptedData/sectionAnomaly/amplitude/README.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
+# Corrupted data - section anomaly in amplitude
+
+Current version: So far, we only have here a gaussian amplitude increase anomaly. 
+
+The initial motivation: consider measuring the access to a website. It varies as the day and night changes. However, if you start a PR campaign	, you expect higher values which in this example are considered as anomalies because the overall tendency is not changed after the end of anomaly. The goal is that the new pattern will not be learnt.
```
Comment (breznak) [inline data/datasets/corruptedData/sectionAnomaly/amplitude/README.md]: whitespace after "PR campaign"

File: opf/anomalyTester.py [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+import os.path
+import csv
+from nupic.algorithms import anomaly_likelihood as al
+
+# GENERAL PRESET
+
+HEADERS = ["time_stamp", "function", "phase", "has_anomaly"]
+
+
+class AnomalyTester:
```
Comment (breznak) [inline opf/anomalyTester.py:32]: add high-level doc what this does?

File: opf/testerI.py [hunk 11]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env python
+
+class TesterI(object):
```
Comment (breznak) [inline opf/testerI.py]: need this? :wink:

**Comment by breznak** [comment]
@wattik had a few minor comments, basically it's just wording; 
I'd like to include the example images directly into the Readme files
Open and describe a separate issue for the "non-integer fs/f ratio" problem

**Comment by breznak** [comment]
@wattik ping? Please fix the typos so we can merge. I can then help you run the remaining results, change to the "timestamp", etc.. this is blocking that.

File: data/datasets/corruptedData/sectionAnomaly/README.md (hunk 8 above)
Comment (wattik) [inline data/datasets/corruptedData/sectionAnomaly/README.md:2]: I was wondering whether we should include this in readme.md in `.../sectionAnomaly/` folder. It seems more logical to me.

Comment (wattik) [inline data/datasets/corruptedData/sectionAnomaly/README.md:2]: Added to `.../sectionAnomaly/`.

**Comment by breznak** [comment]
@wattik great :+1: merging, the only thing left to do is the PNG images.

**Comment by wattik** [comment]
Working on images :)

**Pull request merged**